code_annotations/find_annotations.py

```diff
@@ -0,0 +1,253 @@
+
+        if ext_results:
+            return ext.name, ext_results
+
+    return ext.name, None
+
+
+def format_file_results(source_path, all_results, results):
+    """
+    Add all extensions' search results for a file to the overall results.
+
+    :param source_path: Top level search path
+    :param all_results: Aggregated results to add the results to
+    :param results: Results of search() on a single file
+    :return: None, modifies all_results
+    """
+    # _ here is the extension name, as required by Stevedore map(). Each
+    # annotation already has the extension name so we can ignore it here
+    for _, annotations in results:
+        if annotations is None:
+            continue
+
+        # TODO: The file_path should be the same for all of these results
+        # so we should be able to optimize getting file_path and making
+        # sure it exists in the dict to do this less often.
+
+        # source_path is unnecessary here, we only want the file name
+        # relative to it.
+        file_path = annotations[0]['filename'].replace(source_path, '')
+
+        if file_path not in all_results:
+            all_results[file_path] = []
+
+        # TODO: add support for multiple extensions in the 'found_by' key
+        # and de-dupe results
+        all_results[file_path].extend(annotations)
+
+
+def configure(config, source_path, report_path, verbosity):
+    """
+    Read the configuration file, and handle command line overrides.
+
+    :param config: Location of the configuration file
+    :param source_path: Path to the code to be searched
+    :param report_path: Directory where the report will be generated
+    :param verbosity: Bool indicating verbosity level
+    :return: Configuration dict, updated with overrides
+    """
+    # TODO: Add include / exclude directories
+    ECHO.echo('Reading configuration from {}'.format(config))
+
+    with open(config) as config_file:
+        config = yaml.load(config_file)
+
+    if not source_path and not config['source_path']:
+        raise ConfigurationException('source_path not given and not in configuration file')
+
+    if not report_path and not config['report_path']:
+        raise ConfigurationException('report_path not given and not in configuration file')
+
+    if source_path:
+        config['source_path'] = source_path
+
+    if report_path:
+        config['report_path'] = report_path
+
+    # This is a runtime option, shouldn't be in the config file
+    config['verbosity'] = verbosity
+
+    ECHO.set_verbosity(verbosity)
+    ECHO.echo_v("Verbosity level set to {}".format(config['verbosity']))
+    ECHO.echo_v("Configuration:")
+    ECHO.echo_v(config)
+
+    return config
+
+
+def search(mgr, config):
+    """
+    Walk the source tree, send known file types to extensions.
+
+    :param mgr: Stevedore NamedExtensionManager
+    :param config: Configuration dict
+    :return: Dict containing all of the search results
```

Comment:
I just realized you're not using the google style for docstrings.  Sphinx will parse our normal docstring style just fine, and our cookiecutter even installs `sphinx.ext.napoleon` just to make that work.